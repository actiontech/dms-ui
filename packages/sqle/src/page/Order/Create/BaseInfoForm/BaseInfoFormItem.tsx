import { BasicInput } from '@actiontech/shared';
import {
  FormInputBotBorder,
  FormItemNoLabel,
  FormItemLabel
} from '@actiontech/shared/lib/components/FormCom';
import { workflowNameRule } from '@actiontech/shared/lib/utils/FormRule';
import { ReactNode } from 'react';
import { useTranslation } from 'react-i18next';

const BaseInfoFormItem: React.FC<{ slot?: ReactNode }> = ({ slot }) => {
  const { t } = useTranslation();

  return (
    <>
      <FormItemNoLabel
        name="workflow_subject"
        rules={[
          {
            required: true,
            message: t('common.form.placeholder.input', {
              name: t('order.baseInfo.name')
            })
          },
          {
            validator: workflowNameRule()
          },
          {
            max: 59
          }
        ]}
      >
        <FormInputBotBorder
          className="workflow-name-input-wrapper"
          placeholder={t('common.form.placeholder.input', {
            name: t('order.baseInfo.name')
          })}
        />
      </FormItemNoLabel>

      {slot}

      <FormItemLabel
        className="order-base-info-desc-form-item"
        name="desc"
        label={t('order.baseInfo.describe')}
      >
        <BasicInput.TextArea
          autoSize={{
            maxRows: 10,
            minRows: 8
          }}
          placeholder={t('order.baseInfo.describePlaceholder')}
          maxLength={3000}
          showCount
        />
      </FormItemLabel>
    </>
  );
};

export default BaseInfoFormItem;
