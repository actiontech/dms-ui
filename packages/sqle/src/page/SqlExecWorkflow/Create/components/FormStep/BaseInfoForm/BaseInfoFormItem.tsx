import { BasicInput } from '@actiontech/dms-kit';
import {
  FormInputBotBorder,
  FormItemNoLabel,
  FormItemLabel
} from '@actiontech/dms-kit';
import { workflowNameRule } from '@actiontech/dms-kit';
import { ReactNode, forwardRef } from 'react';
import { useTranslation } from 'react-i18next';
const BaseInfoFormItem = forwardRef<
  HTMLElement,
  {
    slot?: ReactNode;
  }
>(({ slot }, ref) => {
  const { t } = useTranslation();
  return (
    <>
      <section ref={ref}>
        <FormItemNoLabel
          name="workflow_subject"
          rules={[
            {
              required: true,
              message: t('common.form.placeholder.input', {
                name: t('execWorkflow.create.form.baseInfo.name')
              })
            },
            {
              validator: workflowNameRule()
            },
            {
              max: 59
            }
          ]}
        >
          <FormInputBotBorder
            className="workflow-name-input-wrapper"
            placeholder={t('common.form.placeholder.input', {
              name: t('execWorkflow.create.form.baseInfo.name')
            })}
          />
        </FormItemNoLabel>
      </section>

      {slot}

      <FormItemLabel
        className="workflow-base-info-desc-form-item"
        name="desc"
        label={t('execWorkflow.create.form.baseInfo.describe')}
      >
        <BasicInput.TextArea
          autoSize={{
            maxRows: 10,
            minRows: 8
          }}
          placeholder={t(
            'execWorkflow.create.form.baseInfo.describePlaceholder'
          )}
          maxLength={3000}
          showCount
        />
      </FormItemLabel>
    </>
  );
});
export default BaseInfoFormItem;
