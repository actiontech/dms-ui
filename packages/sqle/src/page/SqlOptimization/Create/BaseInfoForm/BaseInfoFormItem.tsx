import { useTranslation } from 'react-i18next';
import { workflowNameRule } from '@actiontech/shared/lib/utils/FormRule';
import { FormInputBotBorder, FormItemNoLabel } from '@actiontech/shared';

const BaseInfoFormItem = () => {
  const { t } = useTranslation();

  return (
    <>
      <FormItemNoLabel
        name="optimizationName"
        rules={[
          {
            required: true,
            message: t('common.form.placeholder.input', {
              name: t('sqlOptimization.create.base.name')
            })
          },
          {
            validator: workflowNameRule()
          },
          {
            max: 59
          }
        ]}
      >
        <FormInputBotBorder
          className="workflow-name-input-wrapper"
          placeholder={t('common.form.placeholder.input', {
            name: t('sqlOptimization.create.base.name')
          })}
        />
      </FormItemNoLabel>
    </>
  );
};

export default BaseInfoFormItem;
