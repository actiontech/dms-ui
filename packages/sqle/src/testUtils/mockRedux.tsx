import whitelist from '../store/whitelist';
import React from 'react';
import { Provider } from 'react-redux';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import sqlManagement from '../store/sqlManagement';
import sqlManagementException from '../store/sqlManagementException';
import pipeline from '../store/pipeline';

const reducers = combineReducers({
  whitelist,
  sqlManagement,
  sqlManagementException,
  pipeline
});

export const storeFactory = (initStore: Dictionary = {}) => {
  return configureStore({
    reducer: reducers,
    preloadedState: initStore
  });
};

export const CustomProvider: React.FC<{
  initStore: Dictionary;
  children: JSX.Element;
}> = (props) => {
  return (
    <Provider store={storeFactory(props.initStore)}>{props.children}</Provider>
  );
};
