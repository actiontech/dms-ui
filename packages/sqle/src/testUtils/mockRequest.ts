import { AxiosResponse } from 'axios';
import ruleTemplate from '@actiontech/shared/lib/api/sqle/service/rule_template';
import configuration from '@actiontech/shared/lib/api/sqle/service/configuration';
import audit_plan from '@actiontech/shared/lib/api/sqle/service/audit_plan';
import { driverMeta } from '../hooks/useDatabaseType/index.test.data';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData
  };
};

export const failedData = (data?: any, otherData?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
    ...otherData
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data)
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data)
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {}
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData)
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {}
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data, otherData)
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data)
      });
    }, 3000);
  });
};

export const mockDriver = () => {
  const spy = jest.spyOn(configuration, 'getDriversV2');
  spy.mockImplementation(() => resolveThreeSecond(driverMeta));
  return spy;
};

export const AuditPlanTypesData = [
  { type: 'default', desc: '自定义', instance_type: '' },
  { type: 'mysql_slow_log', desc: '慢日志', instance_type: 'MySQL' },
  { type: 'mysql_mybatis', desc: 'Mybatis 扫描', instance_type: 'MySQL' },
  { type: 'mysql_schema_meta', desc: '库表元数据', instance_type: 'MySQL' },
  {
    type: 'ali_rds_mysql_slow_log',
    desc: '阿里RDS MySQL慢日志',
    instance_type: 'MySQL'
  },
  {
    type: 'ali_rds_mysql_audit_log',
    desc: '阿里RDS MySQL审计日志',
    instance_type: 'MySQL'
  },
  { type: 'oracle_top_sql', desc: 'Oracle TOP SQL', instance_type: 'Oracle' },
  { type: 'all_app_extract', desc: '应用程序SQL抓取', instance_type: '' },
  { type: 'tidb_audit_log', desc: 'TiDB审计日志', instance_type: 'TiDB' },
  {
    type: 'ocean_base_for_mysql_mybatis',
    desc: 'Mybatis 扫描',
    instance_type: ''
  },
  { type: 'ocean_base_for_mysql_top_sql', desc: 'Top SQL', instance_type: '' }
];
export const mockUseAuditPlanTypes = () => {
  const spy = jest.spyOn(audit_plan, 'getAuditPlanTypesV1');
  spy.mockImplementation(() => resolveThreeSecond(AuditPlanTypesData));
  return spy;
};

export const mockUseRuleType = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTypeByDBTypeV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        rule_count: 0,
        rule_type: 'DML1'
      },
      {
        rule_count: 3,
        rule_type: 'DML2'
      }
    ])
  );

  return spy;
};
