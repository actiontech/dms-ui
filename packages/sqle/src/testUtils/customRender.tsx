import {
  RenderResult,
  render,
  renderHook as renderHookByReact
} from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
  useLocation
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { storeFactory } from './mockRedux';
import { ConfigProvider, theme as antdTheme } from 'antd';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { mount, shallow } from 'enzyme';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';

import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRouterAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <BrowserRouter>
      <Provider store={storeFactory(initStore)}>{ui}</Provider>
    </BrowserRouter>,
    option
  );
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState)
    }
  } as any);
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHookByReact(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
    </StyledEngineProvider>,
    option
  );
};

export const renderWithThemeAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Provider>,
    option
  );
};

export const renderWithThemeAndServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Router>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <StyledEngineProvider injectFirst>
              <ThemeProvider theme={themeData}>{children}</ThemeProvider>
            </StyledEngineProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: themeData
    }
  });
};

export const shallowWithRouter = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: MemoryRouter
  });
};

export const renderLocationDisplay = (): [
  () => RenderResult,
  () => JSX.Element
] => {
  const LocationComponent = () => {
    const location = useLocation();

    return <div data-testid="location-display">{location.pathname}</div>;
  };

  return [() => renderWithRouter(<LocationComponent />), LocationComponent];
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
