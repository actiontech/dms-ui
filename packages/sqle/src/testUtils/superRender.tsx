import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
