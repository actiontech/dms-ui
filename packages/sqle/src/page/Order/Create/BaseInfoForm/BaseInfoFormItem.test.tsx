import { Form } from 'antd';
import { ReactNode } from 'react';
import { cleanup, act, fireEvent } from '@testing-library/react';
import { renderWithTheme } from '../../../../testUtils/customRender';
import { getBySelector } from '@actiontech/shared/lib/testUtil/customQuery';

import BaseInfoFormItem from './BaseInfoFormItem';

describe('sqle/Order/CreateOrder/BaseInfoFormItem', () => {
  const customRender = (params: { slot?: ReactNode } = {}) => {
    return renderWithTheme(
      <Form>
        <BaseInfoFormItem slot={params?.slot} />
      </Form>
    );
  };

  beforeEach(() => {
    jest.useFakeTimers();
    // ignore error: Warning: `NaN` is an invalid value for the `height` css style property.
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
    (console.error as jest.Mock).mockRestore();
    cleanup();
  });

  it('render snap when render default item', () => {
    const { baseElement } = customRender();
    expect(baseElement).toMatchSnapshot();
  });

  it('render workflow_subject check validator', async () => {
    const { baseElement } = customRender();

    const subjectEle = getBySelector('#workflow_subject', baseElement);

    fireEvent.input(subjectEle, {
      target: {
        value: '1234'
      }
    });
    await act(async () => jest.advanceTimersByTime(1000));
    expect(baseElement).toMatchSnapshot();

    fireEvent.change(subjectEle, {
      target: {
        value: '='
      }
    });
    await act(async () => jest.advanceTimersByTime(1000));
    expect(baseElement).toMatchSnapshot();
  });

  it('render when has slot', () => {
    const { baseElement } = customRender({ slot: <span>111</span> });
    expect(baseElement).toMatchSnapshot();
  });
});
