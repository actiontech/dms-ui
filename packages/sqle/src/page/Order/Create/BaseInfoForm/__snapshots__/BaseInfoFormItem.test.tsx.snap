// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqle/Order/CreateOrder/BaseInfoFormItem render snap when render default item 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="ant-form-item css-15d6abf css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class="ant-form-item-required"
              for="workflow_subject"
              title=""
            />
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input css-dev-only-do-not-override-ywkr0o basic-input-wrapper workflow-name-input-wrapper css-opeiev"
                  id="workflow_subject"
                  placeholder="请输入工单名称"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item order-base-info-desc-form-item css-1jlm9cy css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class=""
              for="desc"
              title="工单描述"
            >
              工单描述
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <span
                  class="ant-input-affix-wrapper basic-input-wrapper css-1pd1cd5 ant-input-textarea-affix-wrapper ant-input-textarea-show-count css-dev-only-do-not-override-txh9fw ant-input-show-count"
                  data-count="0 / 3000"
                >
                  <textarea
                    class="ant-input css-dev-only-do-not-override-txh9fw"
                    id="desc"
                    placeholder="点此添加工单描述（选填）"
                    style="overflow-y: hidden; resize: none;"
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="ant-input-data-count"
                    >
                      0 / 3000
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
  <textarea
    aria-hidden="true"
    style="letter-spacing:normal;line-height:1.5714285714285714;padding-top:4px;padding-bottom:4px;font-family:-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';font-weight:;font-size:14px;font-variant:;text-rendering:auto;text-transform:none;width:100%;text-indent:0;padding-left:11px;padding-right:11px;border-width:;box-sizing:border-box;word-break:;white-space:pre-wrap;
  min-height:0 !important;
  max-height:none !important;
  height:0 !important;
  visibility:hidden !important;
  overflow:hidden !important;
  position:absolute !important;
  z-index:-1000 !important;
  top:0 !important;
  right:0 !important;
  pointer-events: none !important;
"
    tab-index="-1"
  />
</body>
`;

exports[`sqle/Order/CreateOrder/BaseInfoFormItem render when has slot 1`] = `
<body>
  <textarea
    aria-hidden="true"
    style="letter-spacing:normal;line-height:1.5714285714285714;padding-top:4px;padding-bottom:4px;font-family:-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';font-weight:;font-size:14px;font-variant:;text-rendering:auto;text-transform:none;width:100%;text-indent:0;padding-left:11px;padding-right:11px;border-width:;box-sizing:border-box;word-break:;white-space:pre-wrap;
  min-height:0 !important;
  max-height:none !important;
  height:0 !important;
  visibility:hidden !important;
  overflow:hidden !important;
  position:absolute !important;
  z-index:-1000 !important;
  top:0 !important;
  right:0 !important;
  pointer-events: none !important;
"
    tab-index="-1"
  />
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="ant-form-item css-15d6abf css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class="ant-form-item-required"
              for="workflow_subject"
              title=""
            />
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input css-dev-only-do-not-override-ywkr0o basic-input-wrapper workflow-name-input-wrapper css-opeiev"
                  id="workflow_subject"
                  placeholder="请输入工单名称"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <span>
        111
      </span>
      <div
        class="ant-form-item order-base-info-desc-form-item css-1jlm9cy css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class=""
              for="desc"
              title="工单描述"
            >
              工单描述
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <span
                  class="ant-input-affix-wrapper basic-input-wrapper css-1pd1cd5 ant-input-textarea-affix-wrapper ant-input-textarea-show-count css-dev-only-do-not-override-txh9fw ant-input-show-count"
                  data-count="0 / 3000"
                >
                  <textarea
                    class="ant-input css-dev-only-do-not-override-txh9fw"
                    id="desc"
                    placeholder="点此添加工单描述（选填）"
                    style="overflow-y: hidden; resize: none;"
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="ant-input-data-count"
                    >
                      0 / 3000
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</body>
`;

exports[`sqle/Order/CreateOrder/BaseInfoFormItem render workflow_subject check validator 1`] = `
<body>
  <textarea
    aria-hidden="true"
    style="letter-spacing:normal;line-height:1.5714285714285714;padding-top:4px;padding-bottom:4px;font-family:-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';font-weight:;font-size:14px;font-variant:;text-rendering:auto;text-transform:none;width:100%;text-indent:0;padding-left:11px;padding-right:11px;border-width:;box-sizing:border-box;word-break:;white-space:pre-wrap;
  min-height:0 !important;
  max-height:none !important;
  height:0 !important;
  visibility:hidden !important;
  overflow:hidden !important;
  position:absolute !important;
  z-index:-1000 !important;
  top:0 !important;
  right:0 !important;
  pointer-events: none !important;
"
    tab-index="-1"
  />
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="ant-form-item css-15d6abf css-dev-only-do-not-override-txh9fw ant-form-item-has-error"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class="ant-form-item-required"
              for="workflow_subject"
              title=""
            />
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-describedby="workflow_subject_help"
                  aria-invalid="true"
                  aria-required="true"
                  class="ant-input ant-input-status-error css-dev-only-do-not-override-ywkr0o basic-input-wrapper workflow-name-input-wrapper css-opeiev"
                  id="workflow_subject"
                  placeholder="请输入工单名称"
                  type="text"
                  value="1234"
                />
              </div>
            </div>
            <div
              style="display: flex; flex-wrap: nowrap;"
            >
              <div
                style="width: 0px; height: 24px;"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-form-item-margin-offset"
          style="margin-bottom: -24px;"
        />
      </div>
      <div
        class="ant-form-item order-base-info-desc-form-item css-1jlm9cy css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class=""
              for="desc"
              title="工单描述"
            >
              工单描述
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <span
                  class="ant-input-affix-wrapper basic-input-wrapper css-1pd1cd5 ant-input-textarea-affix-wrapper ant-input-textarea-show-count css-dev-only-do-not-override-txh9fw ant-input-show-count"
                  data-count="0 / 3000"
                >
                  <textarea
                    class="ant-input css-dev-only-do-not-override-txh9fw"
                    id="desc"
                    placeholder="点此添加工单描述（选填）"
                    style="overflow-y: hidden; resize: none;"
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="ant-input-data-count"
                    >
                      0 / 3000
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</body>
`;

exports[`sqle/Order/CreateOrder/BaseInfoFormItem render workflow_subject check validator 2`] = `
<body>
  <textarea
    aria-hidden="true"
    style="letter-spacing:normal;line-height:1.5714285714285714;padding-top:4px;padding-bottom:4px;font-family:-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';font-weight:;font-size:14px;font-variant:;text-rendering:auto;text-transform:none;width:100%;text-indent:0;padding-left:11px;padding-right:11px;border-width:;box-sizing:border-box;word-break:;white-space:pre-wrap;
  min-height:0 !important;
  max-height:none !important;
  height:0 !important;
  visibility:hidden !important;
  overflow:hidden !important;
  position:absolute !important;
  z-index:-1000 !important;
  top:0 !important;
  right:0 !important;
  pointer-events: none !important;
"
    tab-index="-1"
  />
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-txh9fw"
    >
      <div
        class="ant-form-item css-15d6abf css-dev-only-do-not-override-txh9fw ant-form-item-has-error"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class="ant-form-item-required"
              for="workflow_subject"
              title=""
            />
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-describedby="workflow_subject_help"
                  aria-invalid="true"
                  aria-required="true"
                  class="ant-input ant-input-status-error css-dev-only-do-not-override-ywkr0o basic-input-wrapper workflow-name-input-wrapper css-opeiev"
                  id="workflow_subject"
                  placeholder="请输入工单名称"
                  type="text"
                  value="="
                />
              </div>
            </div>
            <div
              style="display: flex; flex-wrap: nowrap;"
            >
              <div
                style="width: 0px; height: 24px;"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-form-item-margin-offset"
          style="margin-bottom: -24px;"
        />
      </div>
      <div
        class="ant-form-item order-base-info-desc-form-item css-1jlm9cy css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-txh9fw"
          >
            <label
              class=""
              for="desc"
              title="工单描述"
            >
              工单描述
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-txh9fw"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <span
                  class="ant-input-affix-wrapper basic-input-wrapper css-1pd1cd5 ant-input-textarea-affix-wrapper ant-input-textarea-show-count css-dev-only-do-not-override-txh9fw ant-input-show-count"
                  data-count="0 / 3000"
                >
                  <textarea
                    class="ant-input css-dev-only-do-not-override-txh9fw"
                    id="desc"
                    placeholder="点此添加工单描述（选填）"
                    style="overflow-y: hidden; resize: none;"
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="ant-input-data-count"
                    >
                      0 / 3000
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</body>
`;
